libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraComponent.kt

```diff
@@ -1,147 +1,177 @@
                 }
-                when (it.cameraTransition) {
-                    is CameraTransition.ToFollowing -> {
-                        navigationCamera.requestNavigationCameraToFollowing()
-                    }
-                    is CameraTransition.ToOverview -> {
-                        navigationCamera.requestNavigationCameraToOverview()
-                    }
-                    is CameraTransition.ToIdle -> {
-                        navigationCamera.requestNavigationCameraToIdle()
+                viewportDataSource.overviewPadding = overviewEdgeInsets
+                viewportDataSource.followingPadding = followingEdgeInsets
+                viewportDataSource.evaluate()
+            }
+        }
+
+        coroutineScope.launch {
+            locationViewModel.state.collect { location ->
+                location?.let {
+                    viewportDataSource.onLocationChanged(it)
+                    viewportDataSource.evaluate()
+                    if (!cameraViewModel.state.value.isCameraInitialized) {
+                        navigationCamera.requestNavigationCameraToFollowing(
+                            stateTransitionOptions = NavigationCameraTransitionOptions.Builder()
+                                .maxDuration(0) // instant transition
+                                .build()
+                        )
+                        cameraViewModel.invoke(
+                            CameraAction.InitializeCamera(TargetCameraMode.Following)
+                        )
                     }
                 }
             }
         }
+
         coroutineScope.launch {
             mapboxNavigation.flowRouteProgress().collect { routeProgress ->
                 viewportDataSource.onRouteProgressChanged(routeProgress)
                 viewportDataSource.evaluate()
             }
         }
+
         coroutineScope.launch {
-            mapboxNavigation.flowRoutesUpdated().collect { result ->
-                if (result.routes.isEmpty()) {
-                    viewportDataSource.clearRouteData()
+            mapboxNavigation.flowRoutesUpdated().collect {
+                if (it.navigationRoutes.isNotEmpty()) {
+                    viewportDataSource.onRouteChanged(it.navigationRoutes.first())
+                    when (dropInNavigationViewContext.viewModel.navigationState.value) {
+                        NavigationState.ActiveNavigation,
+                        NavigationState.Arrival -> {
+                            cameraViewModel.invoke(CameraAction.ToFollowing)
+                        }
+                        else -> {
+                            cameraViewModel.invoke(CameraAction.ToOverview)
+                        }
+                    }
                 } else {
-                    viewportDataSource.onRouteChanged(result.routes.first())
+                    viewportDataSource.clearRouteData()
+                    cameraViewModel.invoke(CameraAction.ToIdle)
```

Comment:
We don't need to stop flowing when the route is cleared, do we? Let's remain in the state that we are, the nav camera will adapt the zoom level etc.